AirCasting/CreateSessionViews/ConnectingABViewModel.swift

```diff
@@ -0,0 +1,29 @@
+// Created by Lunar on 21/07/2021.
+//
+
+import CoreBluetooth
+import Foundation
+
+protocol ConnectingABProtocol {
```

Comment:
That's a good direction here for sure, you're getting the hang of it! Couple of kinks still to work out here:

`ConnectingABProtocol` - it looks like a Controller/UseCase/Interactor layer object for me. It handles connecting which is our feature here. Great job on spotting it.

The protocol itself doesn't require `bluetoothManager` property to be exposed tho. It's an implementation detail that clients of this protocol won't ever use or have a need to know about. So it should only be required by an implementing class for this.  The other issue with this is separation - now the `ConnectingABProtocol` is depending on a `BluetoothManager` class which is clearly a service layer (it's bluetooth, very low-level detail of communication)

Now for the implementing class you've decided to go with making a `ViewModel` conform to `ConnectingABProtocol`. Remember that our goal with all those protocols and layers is to separate concerns, responsibilities and make classes do only one thing. The job of the ViewModel is simple - get stuff from the controller/usecase/interactor and make it easily digestable by the View. So making `ConnectingABViewModel` conform to `ConnectingABProtocol` denies all this work you've put into separating the two :)

It's always super helpful to spend 2mins on a simple pseudo-uml diagram:

![image](https://user-images.githubusercontent.com/10663361/126482385-d2d92234-b48a-4ca3-877f-450b1b382b66.png)

There are a few problems here with the design. As we can see:
- `ConnectingABViewModel` is conforming to `ConnectingABProtocol` and it's really meant to be using one instead. So we're missing a class that would implement `ConnectingABProtocol` on its own.
- `ConnectingABProtocol` is using `BluetoothManager` directly and classes/interfaces in this layer are prohibited from depending on external stuff, it should define it's expectations and the rest of the app should conform to those.
- `ConnectingABProtocol` knows about bluetooth at all. It should define some connection protocol agnostic of connection type. This way if for example AB4 will have a new connection type (wifi or whatever) we won't have to change any existing code, only introduce a new class for connecting with the new protocol. It also makes unit testing easy and shortens compile times which doesn't seem important now but can get really ugly when the project gets fatter

It should really look like this:

![image](https://user-images.githubusercontent.com/10663361/126483144-0d8eed1e-ae6a-4c5e-a158-d15eaf032c25.png)

But also please rename `ConnectingABProtocol` to something:
- without a `Protocol` in name (it's redundant, everyone knows it's a protocol ;) )
- not using a verb
- perhaps not using AB acronym (AirBeam would be a bit more descriptive)